openmetadata-service/src/main/java/org/openmetadata/service/apps/ConfigurationReader.java

```diff
@@ -0,0 +1,61 @@
+package org.openmetadata.service.apps;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.dropwizard.configuration.ConfigurationException;
+import io.dropwizard.configuration.EnvironmentVariableSubstitutor;
+import io.dropwizard.configuration.FileConfigurationSourceProvider;
+import io.dropwizard.configuration.SubstitutingSourceProvider;
+import io.dropwizard.configuration.YamlConfigurationFactory;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+import org.apache.commons.text.StringSubstitutor;
+import org.openmetadata.schema.api.configuration.apps.AppPrivateConfig;
+import org.openmetadata.service.util.JsonUtils;
+
+public class ConfigurationReader {
+
+  // envMap is for custom environment variables (e.g., for testing), defaulting to the system
+  // environment.
+  private final Map<String, String> envMap;
```

Comment:
ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
    YamlConfigurationFactory<Object> factory =
        new YamlConfigurationFactory<>(Object.class, null, mapper, "dw");
    StringSubstitutor substitutor =
        this.envMap == null
            ? new EnvironmentVariableSubstitutor(false)
            : new StringSubstitutor(this.envMap);
            
            these can be initialize once

Reply:
done